pkg/controller/constraint/constraint_controller.go

```diff
@@ -215,6 +219,26 @@ func (r *ReconcileConstraint) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+func logAddition(l logr.Logger, constraint *unstructured.Unstructured, enforcementAction util.EnforcementAction) {
+	l.Info(
+		"constraint added to OPA",
+		logging.EventType, "constraint_added",
+		logging.ConstraintName, constraint.GetName(),
+		logging.ConstraintAction, string(enforcementAction),
+		logging.ConstraintStatus, "enforced",
+	)
+}
+
+func logRemoval(l logr.Logger, constraint *unstructured.Unstructured, enforcementAction util.EnforcementAction) {
+	l.Info(
+		"constraint removed from OPA",
+		logging.EventType, "constraint_removed",
+		logging.ConstraintName, constraint.GetName(),
+		logging.ConstraintAction, string(enforcementAction),
+		logging.ConstraintStatus, "unenforced",
```

Comment:
this seems very repetitive, can't we do `logAction("constraint removed from OPA", "constraint_removed")` ?

Reply:
Yeah, I kept these separate in case other attributes changed because removal is usually very different from addition/update.

For the CT part of the design, I've done what you suggested, but holding off on submitting that as it conflicts with the pending metrics PR.